### PR TITLE
feat: proposed streetnames can be included in municipality merger

### PR DESCRIPTION
**Pull request by jvandaal**

### CODE REVIEW

File: src/StreetNameRegistry/Municipality/MunicipalityStreetName.cs [hunk 1]
```diff
@@ -26,11 +26,21 @@ public void Approve()
 
         public void ApproveForMunicipalityMerger()
         {
+            if (IsRemoved)
```
Comment (jvandaal) [inline src/StreetNameRegistry/Municipality/MunicipalityStreetName.cs:29]: This was added to satisfy an existing test...

File: src/StreetNameRegistry/Municipality/Events/StreetNameWasProposedForMunicipalityMerger.cs [hunk 2]
```diff
@@ -22,6 +22,9 @@ public sealed class StreetNameWasProposedForMunicipalityMerger : IMunicipalityEv
         [EventPropertyDescription("NIS-code (= objectidentificator) van de gemeente aan dewelke de straatnaam is toegewezen.")]
         public string NisCode { get; }
 
+        [EventPropertyDescription("De gewenste status van de straatnaam na het uitvoeren van de gemeentefusie. Mogelijkheden: Current, Proposed en Retired.")]
```
Comment (ArneD) [inline src/StreetNameRegistry/Municipality/Events/StreetNameWasProposedForMunicipalityMerger.cs]: Retired is not an option.

File: src/StreetNameRegistry/Municipality/Municipality_StreetName.cs [hunk 3]
```diff
@@ -80,6 +81,11 @@ public void ProposeStreetName(Names streetNameNames, PersistentLocalId persisten
                 throw new StreetNamePersistentLocalIdAlreadyExistsException();
             }
 
+            if (desiredStatus != StreetNameStatus.Proposed && desiredStatus != StreetNameStatus.Current)
+            {
+                throw new StreetNameHasInvalidDesiredStatusException();
+            }
+
```
Comment (ArneD) [inline src/StreetNameRegistry/Municipality/Municipality_StreetName.cs]: I don't know if this check is really needed, we will lock the municipality before the data is gone out. We don't do a removed check of the previous streetname either. Also very hard to course correct.

**Review by ArneD — APPROVED** [review]

**Pull request merged**